src/Carbon/Carbon.php

```diff
@@ -933,6 +933,27 @@ public static function createFromTimestampUTC($timestamp)
         return new static('@'.$timestamp);
     }
 
+    /**
+     * Make a Carbon instance from given variable if possible.
+     *
+     * Always return a new instance. Parse only strings and only these likely to be dates (skip ISO 8601
+     * intervals and recurrences). Throw an exception for invalid format, but otherwise return null.
+     *
+     * @param mixed $var
+     *
+     * @return static|null
+     */
+    public static function make($var)
+    {
+        if ($var instanceof DateTime || $var instanceof DateTimeInterface) {
+            return static::instance($var);
+        }
+
+        if (is_string($var) && preg_match('/[:-]/', $var)) {
```

Comment:
It drops the support of `"first monday of january"` strings kind and RegExp is a bit more heavy.

Reply:
It'll be tricky to determine if input is of the `first monday of january` kind or rather `3 days 4 hours` without `try ... catch` block.

But as these methods are now externalized only generic `Invalid ISO 8601 specification`/`Invalid constructor parameters` could be then thrown for invalid input. Not very useful.

Even more difficult as `fromString` will not fail for `first monday of january`.

Reply:
OK to drop, if needed users would just have to use `Carbon::parse()` explicitly.